contracts/GroupHealth.sol

```diff
@@ -0,0 +1,178 @@
+        returns (
+            uint256,
+            uint256,
+            uint256,
+            uint256
+        )
+    {
+        return (1, 1, 0, 0);
+    }
+
+    /**
+     * @notice Updates validator group health.
+     * @param group The group to check for.
+     * @return Whether or not the group is valid.
+     */
+    function updateGroupHealth(address group) public returns (bool) {
+        IValidators validators = getValidators();
+
+        if (!validators.isValidatorGroup(group)) {
+            groupsHealth[group] = false;
+            emit GroupHealthUpdated(group, false);
+            return false;
+        }
+
+        (address[] memory members, , , , , uint256 slashMultiplier, ) = validators
+            .getValidatorGroup(group);
+        // check if group has no members
+        if (members.length == 0) {
+            groupsHealth[group] = false;
+            emit GroupHealthUpdated(group, false);
+            return false;
+        }
+        // check for recent slash
+        if (slashMultiplier < 10**24) {
+            groupsHealth[group] = false;
+            emit GroupHealthUpdated(group, false);
+            return false;
+        }
+
+        // check that at least one member is elected.
+        if (areGroupMembersElected(members)) {
+            groupsHealth[group] = true;
+            emit GroupHealthUpdated(group, true);
+            return true;
+        }
+
+        groupsHealth[group] = false;
+        emit GroupHealthUpdated(group, false);
+        return false;
+    }
+
+    /**
+     * @notice Returns the health state of a validator group.
+     * @param group The group to check for.
+     * @return Whether or not the group is valid.
+     */
+    function isValidGroup(address group) public view returns (bool) {
+        return groupsHealth[group];
+    }
+
+    /**
+     * @notice Gets a validator address from the current validator set.
+     * @param index Index of requested validator in the validator set.
+     * @return Address of validator at the requested index.
+     */
+    function validatorSignerAddressFromCurrentSet(uint256 index)
+        internal
+        view
+        virtual
+        returns (address)
+    {
+        return getElection().validatorSignerAddressFromCurrentSet(index);
+    }
+
+    /**
+     * @notice Gets the size of the current elected validator set.
+     * @return Size of the current elected validator set.
+     */
+    function numberValidatorsInCurrentSet() internal view virtual returns (uint256) {
+        return getElection().numberValidatorsInCurrentSet();
+    }
+
+    /**
+     * @notice Checks if any of group members are elected.
+     * @param members All group members of checked validator group.
+     * @return Whether or not any of the group members are elected.
+     */
+    function areGroupMembersElected(address[] memory members) private returns (bool) {
+        for (uint256 j = 0; j < members.length; j++) {
+            membersMappingHelper[members[j]] = true;
```

Comment:
Since we will usually be writing only 1-3 values but we are going through around 150 validators it actually does help. While testing it seems there is a break even around 100 validators.